engine/src/flutter/display_list/dl_storage.cc

```diff
@@ -20,8 +36,8 @@ uint8_t* DisplayListStorage::allocate(size_t needed) {
   if (used_ + needed > allocated_) {
     static_assert(is_power_of_two(kDLPageSize),
                   "This math needs updating for non-pow2.");
-    // Next greater multiple of DL_BUILDER_PAGE.
-    size_t new_size = (used_ + needed + kDLPageSize) & ~(kDLPageSize - 1);
+    // NPOT, with minimum size of kDLPageSize.
+    size_t new_size = NextPowerOfTwoSize(used_ + needed + kDLPageSize);
```

Comment:
Should we have overflow protection? Hmmm, if we get to 63/64 bits something should be hurting so maybe the realloc results will tip us off... They are FML_CHECK'd below.